producers/mysql-producer/src/main/java/com/flipkart/aesop/runtime/producer/eventprocessor/impl/UpdateEventProcessor.java

```diff
@@ -54,13 +53,8 @@ public void process(BinlogEventV4 event, OpenReplicationListener listener) throw
 		LOGGER.debug("Update Event Received : " + event);
 		UpdateRowsEvent updateRowsEvent = (UpdateRowsEvent) event;
 		List<Pair<Row>> listOfPairs = updateRowsEvent.getRows();
-		List<Row> rowList = new ArrayList<Row>(listOfPairs.size());
-		for (Pair<Row> pair : listOfPairs)
-		{
-			Row row = pair.getAfter();
-			rowList.add(row);
-		}
-		manager.performChanges(updateRowsEvent.getTableId(), updateRowsEvent.getHeader(), rowList, DbusOpcode.UPSERT);
-		LOGGER.debug("Update Successful for  " + event.getHeader().getEventLength() + " . Data updated : " + rowList);
+
+		manager.performChanges(updateRowsEvent.getTableId(), updateRowsEvent.getHeader(), listOfPairs, DbusOpcode.UPSERT);
+		LOGGER.debug("Update Successful for  " + event.getHeader().getEventLength() + " . Data updated : " + listOfPairs);
```

Comment:
Does the SPI support Slf4j like logging functions? This will result in lot of unnecessary String objects getting created.